docs/source/kedro_project_setup/code_beyond_starter_files.md

```diff
@@ -0,0 +1,133 @@
+# Adding code beyond starter files
+
+After you [create a Kedro project](../get_started/new_project.md) and
+[add a pipeline](../tutorial/create_a_pipeline.md), you notice that Kedro generates a
+few boilerplate files: `nodes.py`, `pipeline.py`, `pipeline_registry.py`...
+
+While those may be sufficient for a small project, they quickly become large, hard to
+read and collaborate on as your codebase grows.
+Those files also sometimes make new users think that Kedro requires code
+to be located only in those starter files, which is not true.
```

Comment:
Can you elaborate more on what exact change are you proposing here?
To reference this GH issue right in the `code_beyond_starter_files.md`?
Or to make content of comments on that issue part of this new section of docs?

Reply:
hmm... I am not the best person to ask for English but I'll try my best🤓

> While those may be sufficient for a small project, they quickly become large, hard to read and collaborate on as your codebase grows. Those files also sometimes make new users think that Kedro requires code to be located only in those starter files, which is not true.

I'd rephrase it to something like "When project become large, it may be beneficial to adopt a different structure or give pipeline files a more specific names." It is by convention (and `find_pipeline` use that convention), but not mandatory to name files `pipeline.py`

Reply:
@noklam Your examples is specifically around **_pipeline_** files, while I wanted to convey 2 things here:

- As project evolves and `nodes.py` / `pipeline.py` files grow, they become challenging to manage.
- Good news is that you can rename and restructure them.

Do you disagree with those? I think the first one is just a fact based on how git diff works - if you have one big file, you're more likely to have merge conflicts, etc. And the second one, I wanted to cover not only the pipeline, but also node files there.